docs/usage/BOLFI.rst

```diff
@@ -0,0 +1,309 @@
+details). The ``fit`` method accepts a threshold parameter; if none is
+given, ELFI will use the minimum value of discrepancy estimate mean.
+Afterwards, one may request for a posterior with a different threshold:
+
+.. code:: ipython3
+
+    post2 = bolfi.extract_posterior(-1.)
+
+One can visualize a posterior directly (remember that the priors form a
+triangle):
+
+.. code:: ipython3
+
+    post.plot(logpdf=True)
+
+
+
+.. image:: http://research.cs.aalto.fi/pml/software/elfi/docs/0.5/usage/BOLFI_files/BOLFI_23_0.png
+
+
+Sampling
+~~~~~~~~
+
+Finally, samples from the posterior can be acquired with an MCMC
+sampler. By default it runs 4 chains, and half of the requested samples
+are spent in adaptation/warmup. Note that depending on the smoothness of
+the GP approximation, the number of priors, their gradients etc., this
+may be slow.
+
+.. code:: ipython3
+
+    %time result_BOLFI = bolfi.sample(1000, info_freq=1000)
+
+
+.. parsed-literal::
+
+    INFO:elfi.methods.posteriors:Using optimized minimum value (-1.3110) of the GP discrepancy mean function as a threshold
+    INFO:elfi.methods.mcmc:NUTS: Performing 1000 iterations with 500 adaptation steps.
+    INFO:elfi.methods.mcmc:NUTS: Adaptation/warmup finished. Sampling...
+    INFO:elfi.methods.mcmc:NUTS: Acceptance ratio: 0.404. After warmup 96 proposals were outside of the region allowed by priors and rejected, decreasing acceptance ratio.
+    INFO:elfi.methods.mcmc:NUTS: Performing 1000 iterations with 500 adaptation steps.
+    INFO:elfi.methods.mcmc:NUTS: Adaptation/warmup finished. Sampling...
+    INFO:elfi.methods.mcmc:NUTS: Acceptance ratio: 0.408. After warmup 126 proposals were outside of the region allowed by priors and rejected, decreasing acceptance ratio.
+    INFO:elfi.methods.mcmc:NUTS: Performing 1000 iterations with 500 adaptation steps.
+    INFO:elfi.methods.mcmc:NUTS: Adaptation/warmup finished. Sampling...
+    INFO:elfi.methods.mcmc:NUTS: Acceptance ratio: 0.372. After warmup 99 proposals were outside of the region allowed by priors and rejected, decreasing acceptance ratio.
+    INFO:elfi.methods.mcmc:NUTS: Performing 1000 iterations with 500 adaptation steps.
+    INFO:elfi.methods.mcmc:NUTS: Adaptation/warmup finished. Sampling...
+    INFO:elfi.methods.mcmc:NUTS: Acceptance ratio: 0.461. After warmup 142 proposals were outside of the region allowed by priors and rejected, decreasing acceptance ratio.
+
+
+.. parsed-literal::
+
+    4 chains of 1000 iterations acquired. Effective sample size and Rhat for each parameter:
+    t1 2022.00178671 0.999973259741
+    t2 1806.17269482 1.00236546195
+    CPU times: user 1min 36s, sys: 1.42 s, total: 1min 37s
+    Wall time: 51.7 s
+
+
+The sampling algorithms may be fine-tuned with some parameters. The
+default
+`No-U-Turn-Sampler <http://jmlr.org/papers/volume15/hoffman14a/hoffman14a.pdf>`__
+is a complicated algorithm, and in some cases one may get warnings about
```

Comment:
sophisticated or advanced (complicated seems to have a little negative connotation)